### PR TITLE
Conradomateu day08

### PR DESCRIPTION
**Pull request by bytemines**
# conradomateu's solution for day 08 challenge

## Description

My solution in python of problem number 08 of the Advent of Code!

## Solution

Here is my set of operations and expressions
```python
ops = {
    "inc": operator.add,
    "dec": operator.sub,
    ">": operator.gt,
    "<": operator.lt,
    "==": operator.eq,
    "!=": operator.ne,
    ">=": operator.ge,
    "<=": operator.le}
```

Based on the operations described above, you can parse an operation or an expression. Moreover, you can put it together in the `execute` method.

```python
def parse(op):
    dict[op[0]] = ops[op[1]](dict[op[0]], int(op[2]))
    maxs.append(max(dict.values()))

def parser(exp):
    return ops[exp[1]](dict[exp[0]], int(exp[2]))

def execute(op, exp):
    if parser(exp):
        parse(op)
```

## Tests battery

```python
    def test_parse_operation(self):
        solver.dict["c"] = 8
        self.assertEqual(solver.parser(["c", "dec", "5"]), 3)

    def test_parse_expression(self):
        solver.dict["c"] = 5
        self.assertTrue(solver.parser(["c", "==", "5"]))

    def test_parser_should_be_equals_solution(self):
        solver.dict = {}
        sol = solver.sol(['b inc 5 if a > 1', 'a inc 1 if b < 5', 'c dec -10 if a >= 1', 'c inc -20 if c == 10'])
        self.assertEqual(sol, 1)
```

Happy Advent of Code!

### CODE REVIEW

File: conradomateu/day08/day08.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+def initDictVars(arr):
+    vars = set([x.split(' ')[0] for x in arr])
+    for var in vars:
+        dict[var] = 0
+
+
+def sol(input):
+    initDictVars(input)
+    list = [x.split(" if ") for x in input]
```
Comment (tonilopezmr) [inline conradomateu/day08/day08.py:26]: for comprehesions 👍

File: conradomateu/day08/day08Tests.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import unittest
+import conradomateu.day08.day08 as solver
+
+
+class MyTestCase(unittest.TestCase):
+    @classmethod
```
Comment (tonilopezmr) [inline conradomateu/day08/day08Tests.py:6]: I didn't know about this

**Review by tonilopezmr — APPROVED** [review]
Nice tests 🆑 🆑 

Why you use python?

Do you know that @Cotel ❤️  python?

File: conradomateu/day08/day08Tests.py (hunk 2 above)
Comment (Duxy1996) [reply conradomateu/day08/day08Tests.py:6]: It is very useful , I did not know too. I will use It in the nexts problems.

Comment (Cotel) [reply conradomateu/day08/day08Tests.py:6]: And what does this do? Just remove the `self` parameter in the method?

**Review by Cotel — APPROVED** [review]
It's good but you could improve your methods names, they are too similar like `parse` and `parser`

File: conradomateu/day08/day08Tests.py (hunk 2 above)
Comment (bytemines) [reply conradomateu/day08/day08Tests.py:6]: Is like a Set Up for the class to reuse code.

**Pull request merged**